### PR TITLE
feat: Updating storybook

### PR DESCRIPTION
**Pull request by chadstewart**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

Updated Storybook to with current UI elements in project including adding buttons, cards and other UI elements previously not in Storybook, updating existing elements in Storybook to their current iterations and deleting UI elements that no longer appeared in the project. Looking to ensure that I've covered every UI element in the project as I went through added and updated the ones I visually saw on the project. I am still unfamiliar with the project so I might have missed a few. I don't think there was any gotchas that I came across honestly but whatever issue I was having an issue with I tried to represent in the Storybook doc in this pull request.

## Related Tickets & Documents

Feature #952 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Screenshot from 2021-06-17 11-12-15
Screenshot from 2021-06-17 11-12-26
Screenshot from 2021-06-17 11-14-09
Screenshot from 2021-06-17 11-14-17
Screenshot from 2021-06-17 11-15-34

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [x] 📓 docs
- [ ] 🙅 no documentation needed

### CODE REVIEW

**Comment by bdougie** [comment]
@chadstewart thanks for the PR. Can you include a few more things in this PR? 

- Can you include screenshots? This is a requirement for visual changes. A few screenshots to help folks know what to QA?
- Also any gotchas or things unclear that you had to figure out or workaround? This would be nice for future contributors to have in the body of this PR. 
- Can you also add a storybook docs file similar to the [docs/darkmode](https://github.com/open-sauced/open-sauced/blob/main/docs/dark-mode.md). It does not need to be as detailed, but your experience will help future contributors here too.

File: stories/3-Primitives.stories.js [hunk 1]
```diff
@@ -1,58 +1,39 @@
...
-  <React.Fragment>
-    <Gradient>
-      <div style={{height: 1024, width: 498}} />
-    </Gradient>
-  </React.Fragment>
-);
-
 export const ProfileAvatar = () => (
   <Background style={{height: 1024, padding: "10px"}}>
     <Avatar src={face} alt="profile pic" />
```
Comment (bdougie) [inline stories/3-Primitives.stories.js:15]: See this as a rectangle rather than a square. Are you seeing this?

Screen Shot 2021-06-16 at 9 39 25 PM

File: stories/4-Nav.stories.js [hunk 2]
```diff
@@ -13,6 +18,9 @@ export const HomePage = () => (
   <Fragment>
```
Comment (bdougie) [inline stories/4-Nav.stories.js:18]: I am seeing the Hero there twice. 

Screen Shot 2021-06-16 at 9 47 57 PM

**Review by bdougie — COMMENTED** [review]
questions in two parts.

File: stories/3-Primitives.stories.js (hunk 1 above)
Comment (chadstewart) [reply stories/3-Primitives.stories.js:15]: > See this as a rectangle rather than a square. Are you seeing this?
> 
> Screen Shot 2021-06-16 at 9 39 25 PM

Updated this so it is a square instead of a rectangle and will be added upon completion of all the other suggestions.

File: stories/4-Nav.stories.js (hunk 2 above)
Comment (chadstewart) [reply stories/4-Nav.stories.js:18]: Removed the extra category for Hero and moved that into Homepage category and will be added upon completion of all the other suggestions.

**Comment by chadstewart** [comment]
> * Can you also add a storybook docs file similar to the [docs/darkmode](https://github.com/open-sauced/open-sauced/blob/main/docs/dark-mode.md). It does not need to be as detailed, but your experience will help future contributors here too.

Okay, no problem. I actually have a question on this. So all the other categories are pretty self-explanatory to me but I just realized as I was planning out the doc file, I honestly don't know what Primitives are. While I was working, I put the theme mode buttons there just because it seemed appropriate but now I'm wondering if they belong there or if they should be put in Miscellaneous or buttons instead. I'm also wondering if profile picture should be moved as well.

Let me know whenever you are able and thank you!

**Comment by 0-vortex** [comment]
> > * Can you also add a storybook docs file similar to the [docs/darkmode](https://github.com/open-sauced/open-sauced/blob/main/docs/dark-mode.md?rgh-link-date=2021-06-17T11%3A59%3A09Z). It does not need to be as detailed, but your experience will help future contributors here too.
> 
> Okay, no problem. I actually have a question on this. So all the other categories are pretty self-explanatory to me but I just realized as I was planning out the doc file, I honestly don't know what Primitives are. While I was working, I put the theme mode buttons there just because it seemed appropriate but now I'm wondering if they belong there or if they should be put in Miscellaneous or buttons instead. I'm also wondering if profile picture should be moved as well.
> 
> Let me know whenever you are able and thank you!

Correct me if I'm wrong, not sure if this changed in popular opinion but primitives in this context are react-native elements, unstyled cross-platform aware default components like [Text](https://reactnative.dev/docs/text). The concept has been integrated to React, leading to primitive interfaces for the web, some examples I am familiar with:
- https://emotion.sh/docs/@emotion/primitives
- https://www.npmjs.com/package/react-primitives

TL;DR: what a primitive should be, similar to [JS primitives](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) is an immutable component(interface) used as a baseline for other components (Text, Paragraph, etc)

**Comment by chadstewart** [comment]
> TL;DR: what a primitive should be, similar to [JS primitives](https://developer.mozilla.org/en-US/docs/Glossary/Primitive) is an immutable component(interface) used as a baseline for other components (Text, Paragraph, etc)

It really sounds like I should take out what it in Primitives currently and put them in buttons and Miscellaneous. I don't think we have any stories that is showcasing just primitives. Do we need stories that showcase those?

**Comment by bdougie** [comment]
@chadstewart I think you can add everything in the https://github.com/open-sauced/open-sauced/tree/main/src/styles/Typography folder and remove everything else.

**Comment by bdougie** [comment]
@chadstewart I think you can add everything in the https://github.com/open-sauced/open-sauced/tree/main/src/styles/Typography folder and remove everything else.

**Comment by chadstewart** [comment]
Added the Typography to the Primitives category. Quick note though. I noticed that Microfont and Tinyfont have the same size of font.default. Not sure if it's meant to be that way but it was just something I noticed.

**Comment by bdougie** [comment]
> I noticed that Microfont and Tinyfont have the same size of font.default. Not sure if it's meant to be that way but it was just something I noticed.
Wow checks out, and exactly why this contribution is definitely needed. Looks like [Tinyfont](https://github.com/open-sauced/open-sauced/search?q=Tinyfont) is not being used at all and can be omitted/deleted though.

**Comment by chadstewart** [comment]
Trying to update my pull request description but for some reason it won't let me update the comment. The button isn't highlighted. Do I have to just write a new comment?

**Comment by bdougie** [comment]
Are you are saying this dropdown is not showing?

Screen Shot 2021-06-17 at 8 41 57 PM

**Comment by chadstewart** [comment]
> Are you are saying this dropdown is not showing?
> 
> Screen Shot 2021-06-17 at 8 41 57 PM

It was more the green Update Comment was greyed out.

Screenshot from 2021-06-18 06-47-41

I reloaded the page and I was able to update the comment. I think the issue was that I was trying to update the page while your checks was running. Sorry about the trouble, I'm a little new to all of this honestly.

**Comment by bdougie** [comment]
Upon further review, you need org access to edit the OP. You should have gotten an invite to join.

File: docs/storybook.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+<li><b>Button:</b> These are the button elements that appear in the project in various forms. They primarily are the Button component in the project but can also be icons.</li>
+<li><b>Cards:</b> These are the main container elements in the project. Each item represents a live component in their current form in the project.</li>
+<li><b>Primitives: These are the basic styling of base HTML components.</b></li>
+<li><b>Nav:</b> This is the main navigation bar for the project. There are two states, when there is no user logged in and when a user is logged in.</li>
+<li><b>Footer:</b> This represents the various footers for the project.</li>
+<li><b>Homepage:</b> This is the main component for the project homepage and shows the home page in its current form.</li>
+<li><b>Miscellaneous:</b> These are components that currently don't fit neatly into the above categories.</li>
+</ul>
+
+## Making changes to Storybook
```
Comment (bdougie) [inline docs/storybook.md:25]: Are you still working on this? I think this is close to good, just saw this section was empty..

Comment (chadstewart) [reply docs/storybook.md:25]: Oh sorry. It was meant to be a header for the section. I made some changes to the docs to make it a bit more clear and made the necessary formatting changes to the other headings under it.

**Review by bdougie — APPROVED** [review]
Thanks for working through this @chadstewart
